<a href="https://colab.research.google.com/github/Shiv7397/Shiv7397/blob/main/Financial_Budget_Analysis.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Indian Goverment Financial Analysis 2018 to 2022**


In [ ]:
import numpy as np 
import pandas as pd 
import matplotlib.pyplot as plt
import plotly.express as px

# Upload dataset file(CSV) for analysis of Financial Analysis. (In Colab Notebook we upload file using this command but in Jupiter Notebook in Diffrent Way)

In [ ]:
from google.colab import files
uploaded = files.upload()

Saving FB.csv to FB.csv


# After Uploading CSV File I need to read this file for analysis or working on dataset. I read this file using Pandas library

In [ ]:
df = pd.read_csv('FB.csv')

# Now, I getting columns and rows complete information for finding null values in datasets.

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 57 entries, 0 to 56
Data columns (total 6 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   Ministry/Department  57 non-null     object 
 1   FY 2018              55 non-null     float64
 2   FY 2019              55 non-null     float64
 3   FY 2020              55 non-null     float64
 4   FY 2021              55 non-null     float64
 5   FY 2022              56 non-null     float64
dtypes: float64(5), object(1)
memory usage: 2.8+ KB


# After getting information I found some null values in dataset. let's find null values where it loacated using print full dataset.

In [ ]:
df         # df.head --->> for print first 5 rows

Ministry/Department    FY 2018     FY 2019  \
0         MINISTRY OF AGRICULTURE AND FARMERS WELFARE   57600.00   138563.97   
1                                  MINISTRY OF  AYUSH    1626.37     1939.76   
2               MINISTRY OF CHEMICALS AND FERTILISERS   70586.57    80534.16   
3                          MINISTRY OF CIVIL AVIATION    6602.86     4500.00   
4                                    MINISTRY OF COAL     770.91     1159.05   
5                   MINISTRY OF COMMERCE AND INDUSTRY   11391.93    11893.83   
6                          MINISTRY OF COMMUNICATIONS   39551.46    38637.46   
7   MINISTRY OF CONSUMER AFFAIRS, FOOD AND PUBLIC ...  175944.27   194512.71   
8                             MINISTRY OF COOPERATION        NaN         NaN   
9                       MINISTRY OF CORPORATE AFFAIRS     564.15      586.34   
10                                MINISTRY OF CULTURE    2843.32     3042.35   
11                                MINISTRY OF DEFENCE  404364.71   431010.79   
12   MINISTRY OF DEVELOPMENT  OF NORTH EASTERN REGION    3000.00     3000.00   
13                         MINISTRY OF EARTH SCIENCES    1800.00     1901.76   
14                              MINISTRY OF EDUCATION   85010.29    94853.64   
15  MINISTRY OF ELECTRONICS AND INFORMATION TECHNO...    6000.00     6654.00   
16  MINISTRY OF ENVIRONMENT, FORESTS AND CLIMATE C...    2675.42     2954.72   
17                       MINISTRY OF EXTERNAL AFFAIRS   15011.00    17884.78   
18                                MINISTRY OF FINANCE  894706.82  1005985.08   
19  MINISTRY OF FISHERIES, ANIMAL HUSBANDRY AND DA...        NaN     3737.00   
20             MINISTRY OF FOOD PROCESSING INDUSTRIES    1400.00     1196.60   
21              MINISTRY OF HEALTH AND FAMILY WELFARE   54600.00    64559.12   
22                       MINISTRY OF HEAVY INDUSTRIES    1146.17     1389.64   
23                           MINISTRY OF HOME AFFAIRS  107572.79   119025.28   
24              MINISTRY OF HOUSING AND URBAN AFFAIRS   41765.13    48032.17   
25           MINISTRY OF INFORMATION AND BROADCASTING    4088.98     4375.21   
26                             MINISTRY OF JAL SHAKTI   22356.60    28261.59   
27                  MINISTRY OF LABOUR AND EMPLOYMENT    7700.00    11184.09   
28                        MINISTRY OF LAW AND JUSTICE    4905.35     3611.25   
29    MINISTRY OF MICRO, SMALL AND MEDIUM ENTERPRISES    6552.61     7011.29   
30                                  MINISTRY OF MINES    1669.52     1675.55   
31                       MINISTRY OF MINORITY AFFAIRS    4700.00     4700.00   
32               MINISTRY OF NEW AND RENEWABLE ENERGY    5146.63     5254.83   
33                         MINISTRY OF PANCHAYATI RAJ     825.17      871.37   
34                  MINISTRY OF PARLIAMENTARY AFFAIRS      18.86       19.38   
35  MINISTRY OF PERSONNEL, PUBLIC GRIEVANCES AND P...    1559.61     1762.14   
36              MINISTRY OF PETROLEUM AND NATURAL GAS   31100.55    42901.49   
37                               MINISTRY OF PLANNING     339.65      583.40   
38          MINISTRY OF PORTS, SHIPPING AND WATERWAYS    1881.13     1902.56   
39                                  MINISTRY OF POWER   15046.92    15874.82   
40                               MINISTRY OF RAILWAYS   55088.00    68018.67   
41            MINISTRY OF ROAD TRANSPORT AND HIGHWAYS   71000.00    83015.97   
42                      MINISTRY OF RURAL DEVELOPMENT  114915.32   119874.43   
43                 MINISTRY OF SCIENCE AND TECHNOLOGY   12322.28    13056.24   
44  MINISTRY OF SKILL DEVELOPMENT AND ENTREPRENEUR...    3400.00     2989.21   
45         MINISTRY OF SOCIAL JUSTICE AND EMPOWERMENT    8820.00    10089.90   
46  MINISTRY OF STATISTICS AND PROGRAMME IMPLEMENT...    4859.00     5231.06   
47                                  MINISTRY OF STEEL      47.90      241.29   
48                               MINISTRY OF TEXTILES    7147.73     4831.48   
49                                

# I found null values in dataset, But it is not possible to delete from dataset. So I decided replace this null value with '0'.

In [ ]:
df.fillna(0, inplace= True)
df

Ministry/Department    FY 2018     FY 2019  \
0         MINISTRY OF AGRICULTURE AND FARMERS WELFARE   57600.00   138563.97   
1                                  MINISTRY OF  AYUSH    1626.37     1939.76   
2               MINISTRY OF CHEMICALS AND FERTILISERS   70586.57    80534.16   
3                          MINISTRY OF CIVIL AVIATION    6602.86     4500.00   
4                                    MINISTRY OF COAL     770.91     1159.05   
5                   MINISTRY OF COMMERCE AND INDUSTRY   11391.93    11893.83   
6                          MINISTRY OF COMMUNICATIONS   39551.46    38637.46   
7   MINISTRY OF CONSUMER AFFAIRS, FOOD AND PUBLIC ...  175944.27   194512.71   
8                             MINISTRY OF COOPERATION       0.00        0.00   
9                       MINISTRY OF CORPORATE AFFAIRS     564.15      586.34   
10                                MINISTRY OF CULTURE    2843.32     3042.35   
11                                MINISTRY OF DEFENCE  404364.71   431010.79   
12   MINISTRY OF DEVELOPMENT  OF NORTH EASTERN REGION    3000.00     3000.00   
13                         MINISTRY OF EARTH SCIENCES    1800.00     1901.76   
14                              MINISTRY OF EDUCATION   85010.29    94853.64   
15  MINISTRY OF ELECTRONICS AND INFORMATION TECHNO...    6000.00     6654.00   
16  MINISTRY OF ENVIRONMENT, FORESTS AND CLIMATE C...    2675.42     2954.72   
17                       MINISTRY OF EXTERNAL AFFAIRS   15011.00    17884.78   
18                                MINISTRY OF FINANCE  894706.82  1005985.08   
19  MINISTRY OF FISHERIES, ANIMAL HUSBANDRY AND DA...       0.00     3737.00   
20             MINISTRY OF FOOD PROCESSING INDUSTRIES    1400.00     1196.60   
21              MINISTRY OF HEALTH AND FAMILY WELFARE   54600.00    64559.12   
22                       MINISTRY OF HEAVY INDUSTRIES    1146.17     1389.64   
23                           MINISTRY OF HOME AFFAIRS  107572.79   119025.28   
24              MINISTRY OF HOUSING AND URBAN AFFAIRS   41765.13    48032.17   
25           MINISTRY OF INFORMATION AND BROADCASTING    4088.98     4375.21   
26                             MINISTRY OF JAL SHAKTI   22356.60    28261.59   
27                  MINISTRY OF LABOUR AND EMPLOYMENT    7700.00    11184.09   
28                        MINISTRY OF LAW AND JUSTICE    4905.35     3611.25   
29    MINISTRY OF MICRO, SMALL AND MEDIUM ENTERPRISES    6552.61     7011.29   
30                                  MINISTRY OF MINES    1669.52     1675.55   
31                       MINISTRY OF MINORITY AFFAIRS    4700.00     4700.00   
32               MINISTRY OF NEW AND RENEWABLE ENERGY    5146.63     5254.83   
33                         MINISTRY OF PANCHAYATI RAJ     825.17      871.37   
34                  MINISTRY OF PARLIAMENTARY AFFAIRS      18.86       19.38   
35  MINISTRY OF PERSONNEL, PUBLIC GRIEVANCES AND P...    1559.61     1762.14   
36              MINISTRY OF PETROLEUM AND NATURAL GAS   31100.55    42901.49   
37                               MINISTRY OF PLANNING     339.65      583.40   
38          MINISTRY OF PORTS, SHIPPING AND WATERWAYS    1881.13     1902.56   
39                                  MINISTRY OF POWER   15046.92    15874.82   
40                               MINISTRY OF RAILWAYS   55088.00    68018.67   
41            MINISTRY OF ROAD TRANSPORT AND HIGHWAYS   71000.00    83015.97   
42                      MINISTRY OF RURAL DEVELOPMENT  114915.32   119874.43   
43                 MINISTRY OF SCIENCE AND TECHNOLOGY   12322.28    13056.24   
44  MINISTRY OF SKILL DEVELOPMENT AND ENTREPRENEUR...    3400.00     2989.21   
45         MINISTRY OF SOCIAL JUSTICE AND EMPOWERMENT    8820.00    10089.90   
46  MINISTRY OF STATISTICS AND PROGRAMME IMPLEMENT...    4859.00     5231.06   
47                                  MINISTRY OF STEEL      47.90      241.29   
48                               MINISTRY OF TEXTILES    7147.73     4831.48   
49                                

# I can see that it is not easy to understand and analyze the dataset of all years at once, So I decided to create dataframes separately by year.

# Below 5 steps is common for all years 

# 1. Let's create new DataFrame from existing dataset.

In [ ]:
df18 =df[['Ministry/Department', 'FY 2018']]
df18

Ministry/Department    FY 2018
0         MINISTRY OF AGRICULTURE AND FARMERS WELFARE   57600.00
1                                  MINISTRY OF  AYUSH    1626.37
2               MINISTRY OF CHEMICALS AND FERTILISERS   70586.57
3                          MINISTRY OF CIVIL AVIATION    6602.86
4                                    MINISTRY OF COAL     770.91
5                   MINISTRY OF COMMERCE AND INDUSTRY   11391.93
6                          MINISTRY OF COMMUNICATIONS   39551.46
7   MINISTRY OF CONSUMER AFFAIRS, FOOD AND PUBLIC ...  175944.27
8                             MINISTRY OF COOPERATION       0.00
9                       MINISTRY OF CORPORATE AFFAIRS     564.15
10                                MINISTRY OF CULTURE    2843.32
11                                MINISTRY OF DEFENCE  404364.71
12   MINISTRY OF DEVELOPMENT  OF NORTH EASTERN REGION    3000.00
13                         MINISTRY OF EARTH SCIENCES    1800.00
14                              MINISTRY OF EDUCATION   85010.29
15  MINISTRY OF ELECTRONICS AND INFORMATION TECHNO...    6000.00
16  MINISTRY OF ENVIRONMENT, FORESTS AND CLIMATE C...    2675.42
17                       MINISTRY OF EXTERNAL AFFAIRS   15011.00
18                                MINISTRY OF FINANCE  894706.82
19  MINISTRY OF FISHERIES, ANIMAL HUSBANDRY AND DA...       0.00
20             MINISTRY OF FOOD PROCESSING INDUSTRIES    1400.00
21              MINISTRY OF HEALTH AND FAMILY WELFARE   54600.00
22                       MINISTRY OF HEAVY INDUSTRIES    1146.17
23                           MINISTRY OF HOME AFFAIRS  107572.79
24              MINISTRY OF HOUSING AND URBAN AFFAIRS   41765.13
25           MINISTRY OF INFORMATION AND BROADCASTING    4088.98
26                             MINISTRY OF JAL SHAKTI   22356.60
27                  MINISTRY OF LABOUR AND EMPLOYMENT    7700.00
28                        MINISTRY OF LAW AND JUSTICE    4905.35
29    MINISTRY OF MICRO, SMALL AND MEDIUM ENTERPRISES    6552.61
30                                  MINISTRY OF MINES    1669.52
31                       MINISTRY OF MINORITY AFFAIRS    4700.00
32               MINISTRY OF NEW AND RENEWABLE ENERGY    5146.63
33                         MINISTRY OF PANCHAYATI RAJ     825.17
34                  MINISTRY OF PARLIAMENTARY AFFAIRS      18.86
35  MINISTRY OF PERSONNEL, PUBLIC GRIEVANCES AND P...    1559.61
36              MINISTRY OF PETROLEUM AND NATURAL GAS   31100.55
37                               MINISTRY OF PLANNING     339.65
38          MINISTRY OF PORTS, SHIPPING AND WATERWAYS    1881.13
39                                  MINISTRY OF POWER   15046.92
40                               MINISTRY OF RAILWAYS   55088.00
41            MINISTRY OF ROAD TRANSPORT AND HIGHWAYS   71000.00
42                      MINISTRY OF RURAL DEVELOPMENT  114915.32
43                 MINISTRY OF SCIENCE AND TECHNOLOGY   12322.28
44  MINISTRY OF SKILL DEVELOPMENT AND ENTREPRENEUR...    3400.00
45         MINISTRY OF SOCIAL JUSTICE AND EMPOWERMENT    8820.00
46  MINISTRY OF STATISTICS AND PROGRAMME IMPLEMENT...    4859.00
47                                  MINISTRY OF STEEL      47.90
48                               MINISTRY OF TEXTILES    7147.73
49                                MINISTRY OF TOURISM    2150.00
50                         MINISTRY OF TRIBAL AFFAIRS    6000.00
51            MINISTRY OF WOMEN AND CHILD DEVELOPMENT   24700.00
52               MINISTRY OF YOUTH AFFAIRS AND SPORTS    2196.35
53                                 DEPARMENT OF SPACE   10783.42
54  THE PRESIDENT, PARLIAMENT, UNION PUBLIC SERVIC...    1526.14
55                        DEPARTMENT OF ATOMIC ENERGY   13971.41
56  MINISTRY OF WATER RESOURCES, RIVER DEVELOPMENT...    8860.00

# I can see that it is not easy to include all Departments for analysis. For easy understanding I am taking top 10 most funded Ministries/Departments and rest are included in others.

# 2. let’s prepare the data by only selecting the TOP 10 departments and ramaing all the other departments in the other category:

In [ ]:
df18.nlargest(10, ['FY 2018'])

Ministry/Department    FY 2018
18                                MINISTRY OF FINANCE  894706.82
11                                MINISTRY OF DEFENCE  404364.71
7   MINISTRY OF CONSUMER AFFAIRS, FOOD AND PUBLIC ...  175944.27
42                      MINISTRY OF RURAL DEVELOPMENT  114915.32
23                           MINISTRY OF HOME AFFAIRS  107572.79
14                              MINISTRY OF EDUCATION   85010.29
41            MINISTRY OF ROAD TRANSPORT AND HIGHWAYS   71000.00
2               MINISTRY OF CHEMICALS AND FERTILISERS   70586.57
0         MINISTRY OF AGRICULTURE AND FARMERS WELFARE   57600.00
40                               MINISTRY OF RAILWAYS   55088.00

# 3. Again Create New Dataframe and Insert new row (OTHERS) in the dataframe

In [ ]:
df18 = df18.loc[[18,11,7,42,23,14,41,2,0,40],:]
row = {'Ministry/Department':'OTHERS', 'FY 2018':'405424.5'}
df18 = df18.append(row, ignore_index= True)
df18

Ministry/Department    FY 2018
0                                 MINISTRY OF FINANCE  894706.82
1                                 MINISTRY OF DEFENCE  404364.71
2   MINISTRY OF CONSUMER AFFAIRS, FOOD AND PUBLIC ...  175944.27
3                       MINISTRY OF RURAL DEVELOPMENT  114915.32
4                            MINISTRY OF HOME AFFAIRS  107572.79
5                               MINISTRY OF EDUCATION   85010.29
6             MINISTRY OF ROAD TRANSPORT AND HIGHWAYS    71000.0
7               MINISTRY OF CHEMICALS AND FERTILISERS   70586.57
8         MINISTRY OF AGRICULTURE AND FARMERS WELFARE    57600.0
9                                MINISTRY OF RAILWAYS    55088.0
10                                             OTHERS   405424.5

# 4. To see the government's priorities Let's create a bar chart on the above dataframe

In [ ]:
fig_18 = px.bar(df18, x='Ministry/Department', y='FY 2018', text_auto='.2s', title="Ministry wise Fund allotted (in ₹crore) for 2018")
fig_18.show()

# We can see that the finance department is getting the most of the share from the total budget of the government. 
# 5. Now let’s plot this data into a pie chart to have a clear view of the distribution of funds among all the departments

In [ ]:
fig18_pie = px.pie(df18, values='FY 2018', names='Ministry/Department', color='Ministry/Department', color_discrete_sequence=px.colors.sequential.ice, title='Distrubution of fund by % wise for FY 2018')
fig18_pie.show()

# We can see that for the FY 2018 finance department is getting 36.6% of the funds and Defence Ministry getting 16.6% of the fund.

# Above 5 steps Common for reamaing all Years

In [ ]:
df19 =df[['Ministry/Department', 'FY 2019']]
df19

Ministry/Department     FY 2019
0         MINISTRY OF AGRICULTURE AND FARMERS WELFARE   138563.97
1                                  MINISTRY OF  AYUSH     1939.76
2               MINISTRY OF CHEMICALS AND FERTILISERS    80534.16
3                          MINISTRY OF CIVIL AVIATION     4500.00
4                                    MINISTRY OF COAL     1159.05
5                   MINISTRY OF COMMERCE AND INDUSTRY    11893.83
6                          MINISTRY OF COMMUNICATIONS    38637.46
7   MINISTRY OF CONSUMER AFFAIRS, FOOD AND PUBLIC ...   194512.71
8                             MINISTRY OF COOPERATION        0.00
9                       MINISTRY OF CORPORATE AFFAIRS      586.34
10                                MINISTRY OF CULTURE     3042.35
11                                MINISTRY OF DEFENCE   431010.79
12   MINISTRY OF DEVELOPMENT  OF NORTH EASTERN REGION     3000.00
13                         MINISTRY OF EARTH SCIENCES     1901.76
14                              MINISTRY OF EDUCATION    94853.64
15  MINISTRY OF ELECTRONICS AND INFORMATION TECHNO...     6654.00
16  MINISTRY OF ENVIRONMENT, FORESTS AND CLIMATE C...     2954.72
17                       MINISTRY OF EXTERNAL AFFAIRS    17884.78
18                                MINISTRY OF FINANCE  1005985.08
19  MINISTRY OF FISHERIES, ANIMAL HUSBANDRY AND DA...     3737.00
20             MINISTRY OF FOOD PROCESSING INDUSTRIES     1196.60
21              MINISTRY OF HEALTH AND FAMILY WELFARE    64559.12
22                       MINISTRY OF HEAVY INDUSTRIES     1389.64
23                           MINISTRY OF HOME AFFAIRS   119025.28
24              MINISTRY OF HOUSING AND URBAN AFFAIRS    48032.17
25           MINISTRY OF INFORMATION AND BROADCASTING     4375.21
26                             MINISTRY OF JAL SHAKTI    28261.59
27                  MINISTRY OF LABOUR AND EMPLOYMENT    11184.09
28                        MINISTRY OF LAW AND JUSTICE     3611.25
29    MINISTRY OF MICRO, SMALL AND MEDIUM ENTERPRISES     7011.29
30                                  MINISTRY OF MINES     1675.55
31                       MINISTRY OF MINORITY AFFAIRS     4700.00
32               MINISTRY OF NEW AND RENEWABLE ENERGY     5254.83
33                         MINISTRY OF PANCHAYATI RAJ      871.37
34                  MINISTRY OF PARLIAMENTARY AFFAIRS       19.38
35  MINISTRY OF PERSONNEL, PUBLIC GRIEVANCES AND P...     1762.14
36              MINISTRY OF PETROLEUM AND NATURAL GAS    42901.49
37                               MINISTRY OF PLANNING      583.40
38          MINISTRY OF PORTS, SHIPPING AND WATERWAYS     1902.56
39                                  MINISTRY OF POWER    15874.82
40                               MINISTRY OF RAILWAYS    68018.67
41            MINISTRY OF ROAD TRANSPORT AND HIGHWAYS    83015.97
42                      MINISTRY OF RURAL DEVELOPMENT   119874.43
43                 MINISTRY OF SCIENCE AND TECHNOLOGY    13056.24
44  MINISTRY OF SKILL DEVELOPMENT AND ENTREPRENEUR...     2989.21
45         MINISTRY OF SOCIAL JUSTICE AND EMPOWERMENT    10089.90
46  MINISTRY OF STATISTICS AND PROGRAMME IMPLEMENT...     5231.06
47                                  MINISTRY OF STEEL      241.29
48                               MINISTRY OF TEXTILES     4831.48
49                                MINISTRY OF TOURISM     2189.22
50                         MINISTRY OF TRIBAL AFFAIRS     6894.96
51            MINISTRY OF WOMEN AND CHILD DEVELOPMENT    29164.90
52               MINISTRY OF YOUTH AFFAIRS AND SPORTS     2216.92
53                                 DEPARMENT OF SPACE    12473.26
54  THE PRESIDENT, PARLIAMENT, UNION PUBLIC SERVIC...     1593.25
55                        DEPARTMENT OF ATOMIC ENERGY    16925.51
56  MINISTRY OF WATER RESOURCES, RIVER DEVELOPMENT...        0.00

In [ ]:
df19.nlargest(10, ['FY 2019'])

Ministry/Department     FY 2019
18                                MINISTRY OF FINANCE  1005985.08
11                                MINISTRY OF DEFENCE   431010.79
7   MINISTRY OF CONSUMER AFFAIRS, FOOD AND PUBLIC ...   194512.71
0         MINISTRY OF AGRICULTURE AND FARMERS WELFARE   138563.97
42                      MINISTRY OF RURAL DEVELOPMENT   119874.43
23                           MINISTRY OF HOME AFFAIRS   119025.28
14                              MINISTRY OF EDUCATION    94853.64
41            MINISTRY OF ROAD TRANSPORT AND HIGHWAYS    83015.97
2               MINISTRY OF CHEMICALS AND FERTILISERS    80534.16
40                               MINISTRY OF RAILWAYS    68018.67

In [ ]:
df19 = df19.loc[[18,11,7,42,23,14,41,2,0,40],:]
row = {'Ministry/Department':'OTHERS', 'FY 2019':'450954.80'}
df19 = df19.append(row, ignore_index= True)
df19

Ministry/Department     FY 2019
0                                 MINISTRY OF FINANCE  1005985.08
1                                 MINISTRY OF DEFENCE   431010.79
2   MINISTRY OF CONSUMER AFFAIRS, FOOD AND PUBLIC ...   194512.71
3                       MINISTRY OF RURAL DEVELOPMENT   119874.43
4                            MINISTRY OF HOME AFFAIRS   119025.28
5                               MINISTRY OF EDUCATION    94853.64
6             MINISTRY OF ROAD TRANSPORT AND HIGHWAYS    83015.97
7               MINISTRY OF CHEMICALS AND FERTILISERS    80534.16
8         MINISTRY OF AGRICULTURE AND FARMERS WELFARE   138563.97
9                                MINISTRY OF RAILWAYS    68018.67
10                                             OTHERS   450954.80

In [ ]:
fig_19 = px.bar(df19, x='Ministry/Department', y='FY 2019', text_auto='.2s', title="Ministry wise Fund allotted (in ₹crore) for 2019")
fig_19.show()

# We can see that the again finance department is getting the most of the share from the total budget of the government

In [ ]:
fig19_pie = px.pie(df19, values='FY 2019', names='Ministry/Department', color='Ministry/Department', color_discrete_sequence=px.colors.sequential.ice, title='Distrubution of fund by % wise for FY 2019')
fig19_pie.show()

# We can see that for the FY 2019 finance department is getting 36.1% of the funds and Defence Ministry getting 15.5% of the fund.

In [ ]:
df20 =df[['Ministry/Department', 'FY 2020']]
df20

Ministry/Department     FY 2020
0         MINISTRY OF AGRICULTURE AND FARMERS WELFARE   142762.35
1                                  MINISTRY OF  AYUSH     2122.08
2               MINISTRY OF CHEMICALS AND FERTILISERS    71896.92
3                          MINISTRY OF CIVIL AVIATION     3797.71
4                                    MINISTRY OF COAL      882.61
5                   MINISTRY OF COMMERCE AND INDUSTRY    12824.87
6                          MINISTRY OF COMMUNICATIONS    81957.07
7   MINISTRY OF CONSUMER AFFAIRS, FOOD AND PUBLIC ...   124535.43
8                             MINISTRY OF COOPERATION        0.00
9                       MINISTRY OF CORPORATE AFFAIRS      727.62
10                                MINISTRY OF CULTURE     3149.86
11                                MINISTRY OF DEFENCE   471378.00
12   MINISTRY OF DEVELOPMENT  OF NORTH EASTERN REGION     3048.73
13                         MINISTRY OF EARTH SCIENCES     2070.00
14                              MINISTRY OF EDUCATION    99311.52
15  MINISTRY OF ELECTRONICS AND INFORMATION TECHNO...     6899.03
16  MINISTRY OF ENVIRONMENT, FORESTS AND CLIMATE C...     3100.00
17                       MINISTRY OF EXTERNAL AFFAIRS    17346.71
18                                MINISTRY OF FINANCE  1168937.34
19  MINISTRY OF FISHERIES, ANIMAL HUSBANDRY AND DA...     4114.13
20             MINISTRY OF FOOD PROCESSING INDUSTRIES     1232.94
21              MINISTRY OF HEALTH AND FAMILY WELFARE    67111.80
22                       MINISTRY OF HEAVY INDUSTRIES     1514.13
23                           MINISTRY OF HOME AFFAIRS   167250.33
24              MINISTRY OF HOUSING AND URBAN AFFAIRS    50039.90
25           MINISTRY OF INFORMATION AND BROADCASTING     4375.21
26                             MINISTRY OF JAL SHAKTI    30478.49
27                  MINISTRY OF LABOUR AND EMPLOYMENT    12065.49
28                        MINISTRY OF LAW AND JUSTICE     2778.61
29    MINISTRY OF MICRO, SMALL AND MEDIUM ENTERPRISES     7572.20
30                                  MINISTRY OF MINES     1701.40
31                       MINISTRY OF MINORITY AFFAIRS     5029.00
32               MINISTRY OF NEW AND RENEWABLE ENERGY     5753.00
33                         MINISTRY OF PANCHAYATI RAJ      900.94
34                  MINISTRY OF PARLIAMENTARY AFFAIRS       50.52
35  MINISTRY OF PERSONNEL, PUBLIC GRIEVANCES AND P...     1793.54
36              MINISTRY OF PETROLEUM AND NATURAL GAS    42901.00
37                               MINISTRY OF PLANNING      650.00
38          MINISTRY OF PORTS, SHIPPING AND WATERWAYS     1800.00
39                                  MINISTRY OF POWER    15874.82
40                               MINISTRY OF RAILWAYS    72215.63
41            MINISTRY OF ROAD TRANSPORT AND HIGHWAYS    91823.22
42                      MINISTRY OF RURAL DEVELOPMENT   122398.43
43                 MINISTRY OF SCIENCE AND TECHNOLOGY    14473.29
44  MINISTRY OF SKILL DEVELOPMENT AND ENTREPRENEUR...     3002.21
45         MINISTRY OF SOCIAL JUSTICE AND EMPOWERMENT    11428.96
46  MINISTRY OF STATISTICS AND PROGRAMME IMPLEMENT...     5444.00
47                                  MINISTRY OF STEEL      100.00
48                               MINISTRY OF TEXTILES     3514.79
49                                MINISTRY OF TOURISM     2499.83
50                         MINISTRY OF TRIBAL AFFAIRS     7411.00
51            MINISTRY OF WOMEN AND CHILD DEVELOPMENT    30007.10
52               MINISTRY OF YOUTH AFFAIRS AND SPORTS     2826.92
53                                 DEPARMENT OF SPACE    13479.47
54  THE PRESIDENT, PARLIAMENT, UNION PUBLIC SERVIC...     1641.00
55                        DEPARTMENT OF ATOMIC ENERGY    18228.94
56  MINISTRY OF WATER RESOURCES, RIVER DEVELOPMENT...        0.00

In [ ]:
df20.nlargest(10, ['FY 2020'])

Ministry/Department     FY 2020
18                                MINISTRY OF FINANCE  1168937.34
11                                MINISTRY OF DEFENCE   471378.00
23                           MINISTRY OF HOME AFFAIRS   167250.33
0         MINISTRY OF AGRICULTURE AND FARMERS WELFARE   142762.35
7   MINISTRY OF CONSUMER AFFAIRS, FOOD AND PUBLIC ...   124535.43
42                      MINISTRY OF RURAL DEVELOPMENT   122398.43
14                              MINISTRY OF EDUCATION    99311.52
41            MINISTRY OF ROAD TRANSPORT AND HIGHWAYS    91823.22
6                          MINISTRY OF COMMUNICATIONS    81957.07
40                               MINISTRY OF RAILWAYS    72215.63

In [ ]:
df20 = df20.loc[[18,11,23,0,7,42,14,41,6,40],:]
row = {'Ministry/Department':'OTHERS', 'FY 2020':'499660.80'}
df20 = df20.append(row, ignore_index= True)
df20

Ministry/Department     FY 2020
0                                 MINISTRY OF FINANCE  1168937.34
1                                 MINISTRY OF DEFENCE    471378.0
2                            MINISTRY OF HOME AFFAIRS   167250.33
3         MINISTRY OF AGRICULTURE AND FARMERS WELFARE   142762.35
4   MINISTRY OF CONSUMER AFFAIRS, FOOD AND PUBLIC ...   124535.43
5                       MINISTRY OF RURAL DEVELOPMENT   122398.43
6                               MINISTRY OF EDUCATION    99311.52
7             MINISTRY OF ROAD TRANSPORT AND HIGHWAYS    91823.22
8                          MINISTRY OF COMMUNICATIONS    81957.07
9                                MINISTRY OF RAILWAYS    72215.63
10                                             OTHERS   499660.80

In [ ]:
fig_20 = px.bar(df20, x='Ministry/Department', y='FY 2020', text_auto='.2s', title="Ministry wise Fund allotted (in ₹crore) for 2020")
fig_20.show()

we can see that form last 2 years Finance department getting maximum share of total fund. now again see that finance ministry getting maximum share.

In [ ]:
fig20_pie = px.pie(df20, values='FY 2020', names='Ministry/Department', color='Ministry/Department', color_discrete_sequence=px.colors.sequential.ice, title='Distrubution of fund by % wise for FY 2020')
fig20_pie.show()

# We can see that for the FY 2020 finance department is getting 38.4% of the funds and Defence Ministry getting 15.5% of the fund.

In [ ]:
df21 =df[['Ministry/Department', 'FY 2021']]
df21

Ministry/Department     FY 2021
0         MINISTRY OF AGRICULTURE AND FARMERS WELFARE   131531.19
1                                  MINISTRY OF  AYUSH     2970.30
2               MINISTRY OF CHEMICALS AND FERTILISERS    80714.94
3                          MINISTRY OF CIVIL AVIATION     3224.67
4                                    MINISTRY OF COAL      534.88
5                   MINISTRY OF COMMERCE AND INDUSTRY    12768.25
6                          MINISTRY OF COMMUNICATIONS    75265.22
7   MINISTRY OF CONSUMER AFFAIRS, FOOD AND PUBLIC ...   256948.40
8                             MINISTRY OF COOPERATION        0.00
9                       MINISTRY OF CORPORATE AFFAIRS      712.13
10                                MINISTRY OF CULTURE     2687.99
11                                MINISTRY OF DEFENCE   478195.62
12   MINISTRY OF DEVELOPMENT  OF NORTH EASTERN REGION     2658.00
13                         MINISTRY OF EARTH SCIENCES     1897.13
14                              MINISTRY OF EDUCATION    93224.31
15  MINISTRY OF ELECTRONICS AND INFORMATION TECHNO...     9720.66
16  MINISTRY OF ENVIRONMENT, FORESTS AND CLIMATE C...     2869.93
17                       MINISTRY OF EXTERNAL AFFAIRS    18154.73
18                                MINISTRY OF FINANCE  1386273.30
19  MINISTRY OF FISHERIES, ANIMAL HUSBANDRY AND DA...     4322.82
20             MINISTRY OF FOOD PROCESSING INDUSTRIES     1308.66
21              MINISTRY OF HEALTH AND FAMILY WELFARE    73931.77
22                       MINISTRY OF HEAVY INDUSTRIES     1017.08
23                           MINISTRY OF HOME AFFAIRS   166546.94
24              MINISTRY OF HOUSING AND URBAN AFFAIRS    54581.00
25           MINISTRY OF INFORMATION AND BROADCASTING     4071.23
26                             MINISTRY OF JAL SHAKTI    69053.02
27                  MINISTRY OF LABOUR AND EMPLOYMENT    13306.50
28                        MINISTRY OF LAW AND JUSTICE     3229.94
29    MINISTRY OF MICRO, SMALL AND MEDIUM ENTERPRISES    15699.65
30                                  MINISTRY OF MINES     1466.82
31                       MINISTRY OF MINORITY AFFAIRS     4810.77
32               MINISTRY OF NEW AND RENEWABLE ENERGY     5753.00
33                         MINISTRY OF PANCHAYATI RAJ      913.43
34                  MINISTRY OF PARLIAMENTARY AFFAIRS       65.07
35  MINISTRY OF PERSONNEL, PUBLIC GRIEVANCES AND P...     2097.24
36              MINISTRY OF PETROLEUM AND NATURAL GAS    15943.78
37                               MINISTRY OF PLANNING     1062.77
38          MINISTRY OF PORTS, SHIPPING AND WATERWAYS     1702.35
39                                  MINISTRY OF POWER    15322.00
40                               MINISTRY OF RAILWAYS   110054.64
41            MINISTRY OF ROAD TRANSPORT AND HIGHWAYS   118101.00
42                      MINISTRY OF RURAL DEVELOPMENT   133689.50
43                 MINISTRY OF SCIENCE AND TECHNOLOGY    14794.03
44  MINISTRY OF SKILL DEVELOPMENT AND ENTREPRENEUR...     2785.23
45         MINISTRY OF SOCIAL JUSTICE AND EMPOWERMENT    11689.39
46  MINISTRY OF STATISTICS AND PROGRAMME IMPLEMENT...     1409.13
47                                  MINISTRY OF STEEL       39.25
48                               MINISTRY OF TEXTILES     3631.64
49                                MINISTRY OF TOURISM     2026.77
50                         MINISTRY OF TRIBAL AFFAIRS     7524.87
51            MINISTRY OF WOMEN AND CHILD DEVELOPMENT    24435.00
52               MINISTRY OF YOUTH AFFAIRS AND SPORTS     2596.14
53                                 DEPARMENT OF SPACE    13949.09
54  THE PRESIDENT, PARLIAMENT, UNION PUBLIC SERVIC...     1687.57
55                        DEPARTMENT OF ATOMIC ENERGY    18264.89
56  MINISTRY OF WATER RESOURCES, RIVER DEVELOPMENT...        0.00

In [ ]:
df21.nlargest(10, ['FY 2021'])

Ministry/Department     FY 2021
18                                MINISTRY OF FINANCE  1386273.30
11                                MINISTRY OF DEFENCE   478195.62
7   MINISTRY OF CONSUMER AFFAIRS, FOOD AND PUBLIC ...   256948.40
23                           MINISTRY OF HOME AFFAIRS   166546.94
42                      MINISTRY OF RURAL DEVELOPMENT   133689.50
0         MINISTRY OF AGRICULTURE AND FARMERS WELFARE   131531.19
41            MINISTRY OF ROAD TRANSPORT AND HIGHWAYS   118101.00
40                               MINISTRY OF RAILWAYS   110054.64
14                              MINISTRY OF EDUCATION    93224.31
2               MINISTRY OF CHEMICALS AND FERTILISERS    80714.94

In [ ]:
df21 = df21.loc[[18,11,7,23,42,0,41,40,14,2],:]
row = {'Ministry/Department':'OTHERS', 'FY 2021':'527955.80'}
df21 = df21.append(row, ignore_index= True)
df21

Ministry/Department    FY 2021
0                                 MINISTRY OF FINANCE  1386273.3
1                                 MINISTRY OF DEFENCE  478195.62
2   MINISTRY OF CONSUMER AFFAIRS, FOOD AND PUBLIC ...   256948.4
3                            MINISTRY OF HOME AFFAIRS  166546.94
4                       MINISTRY OF RURAL DEVELOPMENT   133689.5
5         MINISTRY OF AGRICULTURE AND FARMERS WELFARE  131531.19
6             MINISTRY OF ROAD TRANSPORT AND HIGHWAYS   118101.0
7                                MINISTRY OF RAILWAYS  110054.64
8                               MINISTRY OF EDUCATION   93224.31
9               MINISTRY OF CHEMICALS AND FERTILISERS   80714.94
10                                             OTHERS  527955.80

In [ ]:
fig_21 = px.bar(df21, x='Ministry/Department', y='FY 2021', text_auto='.2s', title="Ministry wise Fund allotted (in ₹crore) for 2021")
fig_21.show()

In [ ]:
fig21_pie = px.pie(df21, values='FY 2021', names='Ministry/Department', color='Ministry/Department', color_discrete_sequence=px.colors.sequential.ice, title='Distrubution of fund by % wise for FY 2021')
fig21_pie.show()

# We can see that for the FY 2021 finance department is getting 39.8% of the total funds and Defence Ministry getting 13.7% of the total fund.

In [ ]:
df22 =df[['Ministry/Department', 'FY 2022']]
df22

Ministry/Department     FY 2022
0         MINISTRY OF AGRICULTURE AND FARMERS WELFARE   132513.62
1                                  MINISTRY OF  AYUSH     3050.00
2               MINISTRY OF CHEMICALS AND FERTILISERS   107715.38
3                          MINISTRY OF CIVIL AVIATION    10667.00
4                                    MINISTRY OF COAL      393.24
5                   MINISTRY OF COMMERCE AND INDUSTRY    14421.00
6                          MINISTRY OF COMMUNICATIONS   105406.82
7   MINISTRY OF CONSUMER AFFAIRS, FOOD AND PUBLIC ...   217684.46
8                             MINISTRY OF COOPERATION      900.00
9                       MINISTRY OF CORPORATE AFFAIRS      733.02
10                                MINISTRY OF CULTURE     3009.05
11                                MINISTRY OF DEFENCE   525166.15
12   MINISTRY OF DEVELOPMENT  OF NORTH EASTERN REGION     2800.00
13                         MINISTRY OF EARTH SCIENCES     2653.51
14                              MINISTRY OF EDUCATION   104277.72
15  MINISTRY OF ELECTRONICS AND INFORMATION TECHNO...    14300.00
16  MINISTRY OF ENVIRONMENT, FORESTS AND CLIMATE C...     3030.00
17                       MINISTRY OF EXTERNAL AFFAIRS    17250.00
18                                MINISTRY OF FINANCE  1538779.45
19  MINISTRY OF FISHERIES, ANIMAL HUSBANDRY AND DA...     6037.31
20             MINISTRY OF FOOD PROCESSING INDUSTRIES     2941.99
21              MINISTRY OF HEALTH AND FAMILY WELFARE    86200.65
22                       MINISTRY OF HEAVY INDUSTRIES     3306.00
23                           MINISTRY OF HOME AFFAIRS   185776.55
24              MINISTRY OF HOUSING AND URBAN AFFAIRS    76549.46
25           MINISTRY OF INFORMATION AND BROADCASTING     3980.77
26                             MINISTRY OF JAL SHAKTI    86189.00
27                  MINISTRY OF LABOUR AND EMPLOYMENT    16893.68
28                        MINISTRY OF LAW AND JUSTICE     4054.94
29    MINISTRY OF MICRO, SMALL AND MEDIUM ENTERPRISES    21422.00
30                                  MINISTRY OF MINES     1508.00
31                       MINISTRY OF MINORITY AFFAIRS     5020.50
32               MINISTRY OF NEW AND RENEWABLE ENERGY     6900.68
33                         MINISTRY OF PANCHAYATI RAJ      868.57
34                  MINISTRY OF PARLIAMENTARY AFFAIRS       66.40
35  MINISTRY OF PERSONNEL, PUBLIC GRIEVANCES AND P...     2379.14
36              MINISTRY OF PETROLEUM AND NATURAL GAS     8939.86
37                               MINISTRY OF PLANNING      321.42
38          MINISTRY OF PORTS, SHIPPING AND WATERWAYS     1709.50
39                                  MINISTRY OF POWER    16074.74
40                               MINISTRY OF RAILWAYS   140367.13
41            MINISTRY OF ROAD TRANSPORT AND HIGHWAYS   199107.71
42                      MINISTRY OF RURAL DEVELOPMENT   138203.63
43                 MINISTRY OF SCIENCE AND TECHNOLOGY    14217.46
44  MINISTRY OF SKILL DEVELOPMENT AND ENTREPRENEUR...     2999.00
45         MINISTRY OF SOCIAL JUSTICE AND EMPOWERMENT    13134.93
46  MINISTRY OF STATISTICS AND PROGRAMME IMPLEMENT...     5398.08
47                                  MINISTRY OF STEEL       47.00
48                               MINISTRY OF TEXTILES    12382.14
49                                MINISTRY OF TOURISM     2400.00
50                         MINISTRY OF TRIBAL AFFAIRS     8451.92
51            MINISTRY OF WOMEN AND CHILD DEVELOPMENT    25172.28
52               MINISTRY OF YOUTH AFFAIRS AND SPORTS     3062.60
53                                 DEPARMENT OF SPACE    13700.00
54  THE PRESIDENT, PARLIAMENT, UNION PUBLIC SERVIC...     1649.63
55                        DEPARTMENT OF ATOMIC ENERGY    22723.58
56  MINISTRY OF WATER RESOURCES, RIVER DEVELOPMENT...        0.00

In [ ]:
df22.nlargest(10, ['FY 2022'])

Ministry/Department     FY 2022
18                                MINISTRY OF FINANCE  1538779.45
11                                MINISTRY OF DEFENCE   525166.15
7   MINISTRY OF CONSUMER AFFAIRS, FOOD AND PUBLIC ...   217684.46
41            MINISTRY OF ROAD TRANSPORT AND HIGHWAYS   199107.71
23                           MINISTRY OF HOME AFFAIRS   185776.55
40                               MINISTRY OF RAILWAYS   140367.13
42                      MINISTRY OF RURAL DEVELOPMENT   138203.63
0         MINISTRY OF AGRICULTURE AND FARMERS WELFARE   132513.62
2               MINISTRY OF CHEMICALS AND FERTILISERS   107715.38
6                          MINISTRY OF COMMUNICATIONS   105406.82

In [ ]:
df22 = df22.loc[[18,11,7,42,23,14,41,2,0,40],:]
row = {'Ministry/Department':'OTHERS', 'FY 2022':'654187.80'}
df22 = df22.append(row, ignore_index= True)
df22

Ministry/Department     FY 2022
0                                 MINISTRY OF FINANCE  1538779.45
1                                 MINISTRY OF DEFENCE   525166.15
2   MINISTRY OF CONSUMER AFFAIRS, FOOD AND PUBLIC ...   217684.46
3                       MINISTRY OF RURAL DEVELOPMENT   138203.63
4                            MINISTRY OF HOME AFFAIRS   185776.55
5                               MINISTRY OF EDUCATION   104277.72
6             MINISTRY OF ROAD TRANSPORT AND HIGHWAYS   199107.71
7               MINISTRY OF CHEMICALS AND FERTILISERS   107715.38
8         MINISTRY OF AGRICULTURE AND FARMERS WELFARE   132513.62
9                                MINISTRY OF RAILWAYS   140367.13
10                                             OTHERS   654187.80

In [ ]:
fig_22 = px.bar(df22, x='Ministry/Department', y='FY 2022', text_auto='.2s', title="Ministry wise Fund allotted (in ₹crore) for 2022")
fig_22.show()

In [ ]:
fig22_pie = px.pie(df22, values='FY 2022', names='Ministry/Department', color='Ministry/Department', color_discrete_sequence=px.colors.sequential.ice, title='Distrubution of fund by % wise for FY 2022')
fig22_pie.show()

# We can see that for the FY 2022 finance department is getting 39% of the total funds and Defence Ministry getting 13.3% of the total fund.

# **Conclusion**

# We can see that the which departments  is getting maximum funds from thw total fund as follow -  

1.   MINISTRY OF FINANCE
2.   MINISTRY OF DEFENCE
3.   MINISTRY OF HOME AFFAIRS

# Also Indian Goverment mainly focused on Agriculture, Rural Devlopment, Railways, Road Tansport(Highways) and Education.

#   So this is how we can analyze a dataset that contains data about the revenue and expenditure of the government for a financial years.